STACpopulator/populator_base.py

```diff
@@ -39,6 +42,7 @@ def __init__(
         """
 
         super().__init__()
+        self.configure_app_logging(log_debug)
```

Comment:
Seems odd to me that logging would be configured here, especially since logging can be performed by other modules in the package. 
I think it should be done in the CLI where debug/log-level could be configured/modified by options: https://github.com/crim-ca/stac-populator/blob/10b07161b5690c82d88270bd39ad73e54d02340a/STACpopulator/cli.py#L159

Example of a set of options setup for the CLI: 
https://github.com/crim-ca/weaver/blob/d911c161721c2e25ee4663d34a49d2b218261710/weaver/cli.py#L1785-L1804

Reply:
Doing it in the populator_base allows me to extract the name of the specific derived implementation that is being used, which I then use to name the log file.

Reply:
The file name could be found the same way since the implementation to use is one of the required parameters of the CLI.